test/typechecker_tests.erl

```diff
@@ -96,10 +96,16 @@ subtype_test_() ->
      %% Maps
      ?_assert(subtype(?t( map()             ), ?t( #{a := b}        ))),
      ?_assert(subtype(?t( #{a := b}         ), ?t( map()            ))),
-     ?_assert(subtype(?t( #{a => b}         ), ?t( #{}              ))),
+     ?_assert(subtype(?t( #{}               ), ?t( #{a => b}        ))),
      ?_assert(subtype(?t( #{a := b}         ), ?t( #{a => b}        ))),
-     ?_assert(subtype(?t( #{a => b, c => d} ), ?t( #{a => b}        ))),
+     ?_assert(subtype(?t( #{a => b}         ), ?t( #{a => b, c => d}))),
+     %% TODO: should the below actually hold?
+     %% #{a := b} "forbids" any key but 'a',
+     %% whereas #{a := b, c := d} allows another key 'c'.
+     %% The latter does not really seem like a subtype of the former.
+     %?_assert(subtype(?t( #{a := b, c := d} ), ?t( #{a := b}        ))),
```

Comment:
@zuiderkwast I don't think this should hold, but could you also have a look? It's very similar to the original test which was here on line 101, but I think it actually was buggy.

Reply:
I agree with the comments. It's not a subtype.

Reply:
If you want to fix the test, add _ => _ to the supertype. Then it holds. If this is already covered, just delete it.